src/main/scala/za/co/absa/hyperdrive/trigger/api/rest/services/WorkflowService.scala

```diff
@@ -145,13 +164,22 @@ class WorkflowServiceImpl(override val workflowRepository: WorkflowRepository,
 
   override def runWorkflows(workflowIds: Seq[Long])(implicit ec: ExecutionContext): Future[Boolean] = {
     val userName = getUserName.apply()
-
-    for {
-      joinedWorkflows <- workflowRepository.getWorkflows(workflowIds)
-      dagInstanceJoined <- Future.sequence(joinedWorkflows.map(joinedWorkflow => dagInstanceService.createDagInstance(joinedWorkflow.dagDefinitionJoined, userName)))
-      _ <- dagInstanceRepository.insertJoinedDagInstances(dagInstanceJoined)
-    } yield {
-      true
+    workflowIds.distinct.length match {
+      case numberOfWorkflows if numberOfWorkflows == 1 =>
+        throw new ApiException(GenericError(s"More than 1 workflow has to be triggered!"))
+      case numberOfWorkflows if numberOfWorkflows > ApplicationConfig.maximumNumberOfWorkflowsInBulkRun =>
+        throw new ApiException(GenericError(
+          s"Maximum number of workflows = ${ApplicationConfig.maximumNumberOfWorkflowsInBulkRun}, for bulk run has been exceeded!"
+        ))
+      case numberOfWorkflows =>
+        serviceLogger.debug(s"Bulk run workflows. ${numberOfWorkflows} workflows will be executed.")
```

Comment:
Please also log the username and all workflow ids

Reply:
ok